x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -652,3 +653,46 @@ export const createTotalHitsFromSearchResult = ({
       : searchResult.hits.total.value;
   return totalHits;
 };
+
+interface ReturnType {
+  [timestampString: string]: string[]; // maps timestampString like @timestamp or 'event.ingested' to the indices that contain that timestamp mapping
+}
+
+export const checkMappingForTimestampFields = async (
+  indices: string[],
+  timestamps: string[],
+  services: AlertServices,
+  logger: Logger,
+  buildRuleMessage: BuildRuleMessage
+): Promise<ReturnType> => {
+  try {
+    const foundMappings: GetFieldMappingType = await services.callCluster(
+      'indices.getFieldMapping',
```

Comment:
I think we can use `fieldCaps` here and that does most of the work for us.  Something like:
```
const response = await callWithRequest('fieldCaps', {
      fields: [timestamps],
});
```
For the following query:
```
GET auditbeat-*,logs-*,winlogbeat-*,filebeat-*,packetbeat-*,endgame-*,apm-*-transaction*/_field_caps?fields=event.created,@timestamp&include_unmapped=true
```
That returns us something like:
```
{
  "indices" : [...] // shortening so it fits here,
  "fields" : {
    "@timestamp" : {
      "date" : {
        "type" : "date",
        "searchable" : true,
        "aggregatable" : true
      }
    },
    "event.created" : {
      "date" : {
        "type" : "date",
        "searchable" : true,
        "aggregatable" : true,
        "indices": [...] //shortening so it fits here
      }
    },
    "event" : {
      "object" : {
        "type" : "object",
        "searchable" : false,
        "aggregatable" : false
      }
    }
  }
}
```
So `@timestamp` is missing the `indices` field because it is included in all and mapped the same, but `event.created` does return with an `indices` field and tells us which of ones it appears in mapped the same.

This could catch maybe really unlikely corner cases where the same fields names appear but are mapped differently.

Just a thought... I know we realized we were working on similar things and I'd found this approach to be helpful.

Reply:
ah interesting I'll see about refactoring using this api. Thanks!

Reply:
unfortunately this api is not available with the legacy Elasticsearch client :(